modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -0,0 +1,671 @@
+			$this->migrate_submenus( 'users.php', $users_slug );
+		} else {
+			remove_menu_page( 'profile.php' );
+			remove_submenu_page( 'profile.php', 'grofiles-editor' );
+			remove_submenu_page( 'profile.php', 'grofiles-user-settings' );
+
+			add_menu_page( esc_attr__( 'My Profile', 'jetpack' ), __( 'My Profile', 'jetpack' ), 'read', $profile_slug, null, 'dashicons-admin-users', 70 );
+			add_submenu_page( $profile_slug, esc_attr__( 'Account Settings', 'jetpack' ), __( 'Account Settings', 'jetpack' ), 'read', $account_slug, null, 5 );
+			$this->migrate_submenus( 'profile.php', $profile_slug );
+		}
+	}
+
+	/**
+	 * Adds Tools menu.
+	 *
+	 * @param string $domain  Site domain.
+	 * @param bool   $calypso Optional. Whether links should point to Calypso or wp-admin. Default true (Calypso).
+	 */
+	public function add_tools_menu( $domain, $calypso = true ) {
+		$admin_slug = 'tools.php';
+		$menu_slug  = $calypso ? 'https://wordpress.com/marketing/tools/' . $domain : $admin_slug;
+
+		add_submenu_page( $menu_slug, esc_attr__( 'Marketing', 'jetpack' ), __( 'Marketing', 'jetpack' ), 'manage_options', 'https://wordpress.com/marketing/tools/' . $domain, null, 5 );
+		add_submenu_page( $menu_slug, esc_attr__( 'Earn', 'jetpack' ), __( 'Earn', 'jetpack' ), 'manage_options', 'https://wordpress.com/earn/' . $domain, null, 10 );
+
+		if ( $calypso ) {
+			remove_menu_page( $admin_slug );
+			remove_submenu_page( $admin_slug, $admin_slug );
+			remove_submenu_page( $admin_slug, 'import.php' );
+			remove_submenu_page( $admin_slug, 'export.php' );
+			remove_submenu_page( $admin_slug, 'delete-blog' );
+
+			add_menu_page( esc_attr__( 'Tools', 'jetpack' ), __( 'Tools', 'jetpack' ), 'manage_options', $menu_slug, null, 'dashicons-admin-tools', 75 );
+			add_submenu_page( $menu_slug, esc_attr__( 'Import', 'jetpack' ), __( 'Import', 'jetpack' ), 'import', 'https://wordpress.com/import/' . $domain, null, 15 );
+			add_submenu_page( $menu_slug, esc_attr__( 'Export', 'jetpack' ), __( 'Export', 'jetpack' ), 'export', 'https://wordpress.com/export/' . $domain, null, 20 );
+
+			$this->migrate_submenus( $admin_slug, $menu_slug );
+		}
+	}
+
+	/**
+	 * Adds Settings menu.
+	 *
+	 * @param string $domain Site domain.
+	 * @param bool   $calypso Optional. Whether links should point to Calypso or wp-admin. Default true (Calypso).
+	 */
+	public function add_options_menu( $domain, $calypso = true ) {
+		if ( $calypso ) {
+			remove_submenu_page( 'options-general.php', 'options-discussion.php' );
+			remove_submenu_page( 'options-general.php', 'options-writing.php' );
+		}
+
+		add_options_page( esc_attr__( 'Domains', 'jetpack' ), __( 'Domains', 'jetpack' ), 'manage_options', 'https://wordpress.com/domains/manage/' . $domain, null, 1 );
+		add_options_page( esc_attr__( 'Hosting Configuration', 'jetpack' ), __( 'Hosting Configuration', 'jetpack' ), 'manage_options', 'https://wordpress.com/hosting-config/' . $domain, null, 6 );
```

Comment:
Maybe we should add additional checks for features like those that are clearly Atomic-only, to ensure that they cannot be accessed when your site is not on Atomic?

Reply:
This particular item seems to be available for simple sites as well. Are there menu items besides this one that stand out to you that could benefit from a check?

Reply:
I was more thinking of non-Atomic Jetpack sites in this scenario. They do not have access to those pages.

Reply:
So far I've been operating under the assumption that the admin-menu part will only be accessible on WoA. Should I be coding more defensively and account for the case of Jetpack sites, too?

Reply:
3f57bbb363e5d92d3f42ba89e61eac3bb2999e57